data-raw/ictr/ictr_main.Rmd

```diff
@@ -0,0 +1,293 @@
+output: github_document
+params: 
+  version: "0.0.0.9000"
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Setup
+
+Load Packages
+```{r}
+library(tidyverse, warn.conflicts = FALSE)
+library(vroom)
+library(here)
+```
+
+# Introduction
+
+The Input Product Carbon Transition Risk Indicator assesses the transition risk
+of the input products due to their relative carbon footprint to other input
+products. As a default option, each input product will be compared to the carbon
+footprint of every other input product. Alternatively, users can also choose to
+compare those input products of the same type, so e.g., comparing only raw
+materials with each other. Input products with a higher carbon footprint will
+also face a higher risk.
+
+The Input Product Carbon Transition Risk Indicator is therefore similar to the
+Product Carbon Transition Risk Indicator, but it focuses on the input products
+and not the product of the company. Input products are, for example, resources,
+packaging materials, energy and enabling services (such as tractor use on farm)
+to produce the product.
+
+After identifying each carbon footprint for one input product, the input
+products will be ranked according to their footprint. Input products in the
+highest percentile (≥70%) will be classified as high transition risk input
+products. Input products in the medium percentile (between ≥30% and <70%) will
+be classified as medium transition risk input products. Products in the lowest
+percentile (<30%) will be classified as low transition risk input products.
+
+After assessing the input products transition risk based on its carbon footprint
+for each product, they will be aggregated at company-level. We derive what
+percentage of the input products are high, medium and low transition risk.
+
+The goal of the transition risk MVP is to create a first draft of how the
+transition risk indicator for inputs would be build in code in order to convert
+it into code production easily in the future.
+
+The transition risk indicator consists of 3 main steps:
+
+- Step 1: Identifying the input products for each product
+- Step 2: Calculating the relative carbon footprint per input product
+- Step 3: Aggregating on the company-level
+
+# Step 1: Identifying
+
+Identifying the input products for each product
+
+## 1.1 Load company data and inputs data
+
+Sample data set includes inputs and co2 footprints for each product from
+Ecoinvent and sectors from Europages. NOTE: the following columns are a complete
+random selection and do not reflect the true information:
+
+    - co2 footprints (not allowed to share licensed data right now)
+    - sectors (as the matching with ecoinvent is not done yet, we do not have one sector per product yet)
+
+```{r}
+inputs <- read.csv(here("data_prep","02_input_data", "data_ecoinvent_co2.csv"))
+companies <- read.csv(here("data_prep","02_input_data", "tilt_companies.csv"))
+```
```

Comment:
Instead of creating a new dataset, can we instead reuse [`pctr_companies`](https://2degreesinvesting.github.io/tiltIndicator/reference/pctr_companies.html)?

```r
library(tiltIndicator, warn.conflicts = FALSE)

# They have identical structure
companies |> glimpse()
#> Rows: 25
#> Columns: 6
#> $ company_id            <chr> "fleischerei-stiefsohn_00000005219477-001", "fle…
#> $ company_name          <chr> "fleischerei stiefsohn", "fleischerei stiefsohn"…
#> $ ep_product            <chr> "shellfish", "crustaceans", "fish", "meat", "off…
#> $ activity_product_uuid <chr> "0a242b09-772a-5edf-8e82-9cb4ba52a258_ae39ee61-d…
#> $ ei_activity           <chr> "transport, freight, lorry 7.5-16 metric ton, EU…
#> $ unit                  <chr> "metric ton*km", "kg", "metric ton*km", "kg", "m…
pctr_companies |> glimpse()
#> Rows: 25
#> Columns: 6
#> $ company_id            <chr> "powerstartmanufacturinggmbh_0000000546010082730…
#> $ company_name          <chr> "powerstart_manufacturing_gmbh", "powerstart_man…
#> $ ep_product            <chr> "mining_and_quarrying_machinery_and_equipment", …
#> $ activity_product_uuid <chr> "0a242b09-772a-5edf-8e82-9cb4ba52a258_ae39ee61-d…
#> $ ei_activity           <chr> "transport, freight, lorry 7.5-16 metric ton, EU…
#> $ unit                  <chr> "metric ton*km", "kg", "metric ton*km", "kg", "m…

# Some columns are the same in both datasets
is_same <- function(x) identical(sort(companies[[x]]), sort(pctr_companies[[x]]))

names(companies) |> 
  map(is_same) |> 
  set_names(names(companies))
#> $company_id
#> [1] FALSE
#> 
#> $company_name
#> [1] FALSE
#> 
#> $ep_product
#> [1] FALSE
#> 
#> $activity_product_uuid
#> [1] TRUE
#> 
#> $ei_activity
#> [1] TRUE
#> 
#> $unit
#> [1] TRUE
```